src/modules/user/user.controller.ts

```diff
@@ -161,6 +161,66 @@ export class UserController {
     }
   }
 
+  @UseGuards(ThrottlerGuard)
+  @Throttle({
+    long: {
+      ttl: 3600,
+      limit: 5,
+    },
+  })
+  @Get('/user/:email')
+  @HttpCode(201)
+  @ApiOperation({
+    summary:
+      'Get the user data by email and check if the user has subscription',
+  })
+  @ApiOkResponse({
+    description: 'Get the user data by email',
+  })
+  @ApiBadRequestResponse({ description: 'Missing required fields' })
+  @ApiParam({
+    name: 'email',
+    type: String,
+  })
+  async getUserByEmail(
+    @Param('email') email: User['email'],
+    @Res({ passthrough: true }) res: Response,
+  ) {
+    try {
+      const user = await this.userUseCases.getUserByUsername(email);
+      if (!user) {
+        throw new NotFoundException('PRE_CREATED_USER_NOT_FOUND');
```

Comment:
This is wrong, precreated users are for invited users to a private sharing, this is not related. Just return a NotFound without any message by parameter